### PR TITLE
lburnett/filter hosts by display name [ENT-3195]

### PR DESCRIPTION
**Pull request by lindseyburnett**
Add support for `display_name_contains` as an optional query param for GET `/hosts/products/{product_id}`.

Expanded the jpql `@Query` in HostRepository.  A potential item for future improvement would be to update the related `@Entity` classes and their relationships, and refactor use spring data's `Specification` api to dynamically build the WHERE expressions and not have to explicitly concern ourselves with the optional query params coming into the endpoint being null/empty/converting to "_ANY", etc.

### CODE REVIEW

File: src/main/java/org/candlepin/subscriptions/db/model/Host.java [hunk 1]
```diff
@@ -304,6 +307,20 @@ public void setCloudProvider(String cloudProvider) {
         this.cloudProvider = cloudProvider;
     }
 
+    @PrePersist
+    void prePersist() {
+        if (Objects.isNull(displayName)) {
+            displayName = "";
```
Comment (lindseyburnett) [inline src/main/java/org/candlepin/subscriptions/db/model/Host.java:313]: Theoretically, displayName shouldn't ever be null in the wild.  If somehow it is before a Host entity gets saved, default it to an empty string instead of `null`.  This is mostly to keep from having an overly-complex @Query in `HostRepository` with null & empty string checks for a scenario that doesn't happen.

Eventually we should probably enforce some constraints on the hosts.display_name column to solidify this.

Anyone think it's necessary to also attach a database update for `update hosts set display_name = '' where display_name is null`?  I wouldn't actually expect it to update any records, but figured I'd ask to just cross my T's and dot my I's.

File: src/test/java/org/candlepin/subscriptions/db/HostRepositoryTest.java [hunk 2]
```diff
@@ -539,6 +541,50 @@ void testShouldFilterCores() {
         assertEquals(1, physical.getCores());
     }
 
+    @Transactional
+    @ParameterizedTest
+    @CsvSource({
```
Comment (lindseyburnett) [inline src/test/java/org/candlepin/subscriptions/db/HostRepositoryTest.java:546]: At some point I'd like to look into bumping our JUnit version.  With newer minor versions, there are more options available for parameterized tests - like more defined/obvious ways of passing around null sources.

There's also an option (and available with our current version I think) to use this `@CsvSource` with an external resource file instead.  Anyone have thoughts or preference on that?  Personally, I think it's easier to read with the data in-line like this to prevent the need to open and switch between another file.

File: src/main/java/org/candlepin/subscriptions/db/model/Host.java (hunk 1 above)
Comment (kahowell) [reply src/main/java/org/candlepin/subscriptions/db/model/Host.java:313]: Instead of this, I was thinking we'd just do the equivalent in liquibase:

```xml
<addNotNullConstraint tableName="hosts" columnName="display_name" defaultNullValue="" />
```

Thoughts?

File: src/test/java/org/candlepin/subscriptions/db/HostRepositoryTest.java (hunk 2 above)
Comment (kahowell) [reply src/test/java/org/candlepin/subscriptions/db/HostRepositoryTest.java:546]: I dig in-line here. I'd say a rule of thumb would be appropriate when reviewing. I would go with more than 10 lines of CSV should be in a separate file.

**Review by kahowell — CHANGES_REQUESTED** [review]
Needs a rebase, and I'd like to see nullability changed in liquibase (see inline comment). Looks good otherwise.

File: src/main/resources/liquibase/202012141133-enforce-display-name-not-null.xml [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    <property name="table.name" value="hosts"/>
+    <property name="column.name" value="display_name"/>
+
+    <changeSet id="202012141133" author="lburnett">
+        <comment>
+            Add not null constraint to ${table.name}.${column.name} and set existing null values to empty
+            string
+        </comment>
+        <addNotNullConstraint tableName="${table.name}" columnName="${column.name}" defaultNullValue=""/>
+    </changeSet>
```
Comment (kahowell) [inline src/main/resources/liquibase/202012141133-enforce-display-name-not-null.xml:17]: Can you please inline table.name and column.name?

**Review by kahowell — CHANGES_REQUESTED** [review]
Overall looks good. Tiny request (see inline)

File: src/main/resources/liquibase/202012141133-enforce-display-name-not-null.xml (hunk 3 above)
Comment (lindseyburnett) [reply src/main/resources/liquibase/202012141133-enforce-display-name-not-null.xml]: all set

**Review by kahowell — APPROVED** [review]
